File: doc-examples/jdbc-example-java/src/main/java/example/Product.java
package example;

import javax.persistence.*;

@Entity
public class Product {

    @Id
    @GeneratedValue
    private Long id;
    private String name;
    @ManyToOne
    private Manufacturer manufacturer;

    public Product(String name, Manufacturer manufacturer) {
        this.name = name;
        this.manufacturer = manufacturer;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public Manufacturer getManufacturer() {
        return manufacturer;
    }
}

File: doc-examples/hibernate-example-python/src/main/python/example/Product.py
# tag::entity[]
# tag::entitywithprocedures[]
from dataclasses import dataclass
from typing import Annotated

from jakarta.persistence import Entity, GeneratedValue, Id, ManyToOne, NamedStoredProcedureQuery, StoredProcedureParameter
# end::entity[]
from java.lang import Long

from example.Manufacturer import Manufacturer


@NamedStoredProcedureQuery(name="calculateSum",
                           procedureName="calculateSumInternal",
                           parameters=[
                               StoredProcedureParameter(name="productId", mode="IN", type=Long),
                               StoredProcedureParameter(name="result", mode="OUT", type=Long)
                           ])
# tag::entity[]
@Entity
@dataclass
class Product:
    # end::entitywithprocedures[]
    name: str | None = None
    manufacturer: Annotated[Manufacturer | None, ManyToOne(optional=False, fetch="LAZY")] = None
    id: Annotated[int | None, Id, GeneratedValue] = None
# end::entity[]
